pysmt/rewritings.py

```diff
@@ -618,26 +618,78 @@ def walk_ite(self, formula, args, **kwargs):
         else:
             return formula
 
+
+class TimesDistributor(IdentityDagWalker):
+    """Normalize the use of multiplication by pushing it into the leafs.
+
+    E.g., (x+1)*3 -> (x*3) + 3
```

Comment:
It should, but you can add it as an example to be sure! :) 

It will first distribute on the left x*(x-1)...  That formula will be process immediately after (notice that get_children is called when going down the tree) and thus it will distribute on the right, since there is a symbol  on the ledt.